pyserini/trectools/_base.py

```diff
@@ -99,6 +152,71 @@ def rescore(self, method: RescoreMethod, rrf_k: int = None, scale: float = None)
     def to_numpy(self) -> np.ndarray:
         return self.run_data.to_numpy(copy=True)
 
+    def remove_by_qrels(self, qrels: Qrels, copy=True):
+        """Given a Qrels object, remove each docid in self if docid is also in the qrels.
+        This operation is performed on each topic separately.
+
+        Parameters:
+        ----------
+        qrels : Qrels
+            Qrels with docids to remove from TrecRun.
+        copy : Bool
+            Return a new TrecRun object if True, else self will be modified and returned.
+        """
+
+        return self._filter_from_qrels(qrels, False, copy=copy)
+
+    def keep_only_qrels(self, qrels: Qrels, copy=True):
+        """Given a Qrels object, keep each docid in self if docid is also in the qrels.
+        This operation is performed on each topic separately.
+        After this operation, judged@x based on the given qrels should be 1.
+
+        Parameters:
+        ----------
+        qrels : Qrels
+            Qrels with docids to keep in TrecRun.
+        copy : Bool
+            Return a new TrecRun object if True, else self will be modified and returned.
+        """
+
+        return self._filter_from_qrels(qrels, True, copy=copy)
+
+    def _filter_from_qrels(self, qrels: Qrels, keep: bool, copy=True):
+        """Private helper function to remove/keep each docid in self if docid is also in the given Qrels object.
+        This operation is performed on each topic separately.
+
+        Parameters:
+        ----------
+        qrels : Qrels
+            Qrels with docids to remove from or keep in TrecRun.
+        copy : Bool
+            If True, only docids in qrels will appear in TrecRun. Else, docids in qrels won't appear in the TrecRun.
+        copy : Bool
+            Return a new TrecRun object if True, else self will be modified and returned.
+        """
+
+        df_list = []
+        for topic in self.topics():
+            if topic not in qrels.topics():
+                continue
+
+            tabu_list = qrels.get_docids(topic)
```

Comment:
what's "tabu" mean?

Reply:
Coming from [Tabu Search](https://www.sciencedirect.com/topics/computer-science/tabu-search#:~:text=2.2%20Tabu%20Search%20Algorithm%20(TS,composed%20by%20the%20next%20elements.), tabu just means something we don't want. Maybe not the best naming here. I am going to name it `qrels_docids`.